Automattic/jetpack
# Added an additional GET parameter to profile sandbox requests.

The additional constant enables adding an XDEBUG_PROFILE constant to all sandbox requests to allow for easier profiling.




#### Changes proposed in this Pull Request:
* An additional constant added that sends an extra GET parameter with all sandbox requests.

#### Jetpack product discussion
N/A

#### Does this pull request change what data or activity we track or use?
No tracking changes.

#### Testing instructions:
* Sandbox your Jetpack installation using the `JETPACK__SANDBOX_DOMAIN` constant.
* Enable Xdebug on your sandbox using `sudo xdebug enable`.
* Add a filter to enable the parameter addition
```
add_filter( 'jetpack_sandbox_add_profile_parameter', '__return_true' ); 
```
* The code above will add the parameter to all requests. You can parse the URL and HOST provided to enable it only for certain requests
* Create a post using a CLI command:
```
wp post generate --count=1
```
* Observe the sync request in the output log. Note that the URL listed in the output will be the original one, not the sandboxed one, so it will not have the profile parameter added.
* Observe a cachegrind output file in your sandbox `/tmp` folder. Be careful, these can be quite large.

#### Proposed changelog entry for your changes:

* N/A
